packages/web3-core/types/index.d.ts

```diff
@@ -0,0 +1,62 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+import {AbstractProviderAdapter, ProvidersModuleFactory, provider} from 'web3-providers';
+
+export class AbstractWeb3Module {
+    constructor(
+        provider: AbstractProviderAdapter | provider | string,
+        providersModuleFactory: ProvidersModuleFactory,
+        // not sure what the below object structure is
+        providers: {},
```

Comment:
it is:
```js
{
    HttpProvider: HttpProvider,
    WebsocketProvider: WebsocketProvider,
    IpcProvider: IpcProvider
}
```

as you can see here: https://github.com/ethereum/web3.js/blob/ethereumProvider/packages/web3-providers/src/index.js#L35-L39